trace/tracer.go

```diff
@@ -57,26 +57,16 @@ func New(config *Config) (trace.Tracer, error) {
 		}, nil
 	}
 
-	exporter, err := zipkin.New(
-		"http://localhost:9411/api/v2/spans",
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	tracerProviderOpts := []sdktrace.TracerProviderOption{
-		sdktrace.WithBatcher(exporter, sdktrace.WithExportTimeout(tracerExportTimeout)),
-		sdktrace.WithResource(
-			resource.NewWithAttributes(
-				semconv.SchemaURL,
-				attribute.String("version", config.Version),
-				semconv.ServiceNameKey.String(config.Agent),
-			),
+	uptrace.ConfigureOpentelemetry(
```

Comment:
I'm wondering why we import `uptrace` at all instead of just passing a header into the `sdktrace` options below like AvalancheGo: https://github.com/ava-labs/avalanchego/pull/1727

The appeal there is that we can just use the stock OpenTelemetry?

Reply:
In the process, we should just migrate to use this package (we don't need a separate `trace` package anymore because `uptrace` can use opentelemtry directly ❤️ ): https://github.com/ava-labs/avalanchego/blob/master/trace/tracer.go

Reply:
trace has been updated.

you wanted to do the same with prometheus? delete all reference to prometheus in the code and use opentelemetry metrics instead?

Reply:
trace package has been updated. If you want to use `trace` package from avalanchego we need to pass AppName and Version from the Config.

https://github.com/ava-labs/avalanchego/blob/e70a17d9d988b5067f3ef5c4a057f15ae1271ac4/trace/tracer.go#L73C2-L73C2

https://github.com/ava-labs/avalanchego/blob/e70a17d9d988b5067f3ef5c4a057f15ae1271ac4/trace/tracer.go#L74

Reply:
I think that would be a good change. Can you open an issue for that? We can probably merge that quickly.

Reply:
I think we should try to do so (use the `trace` package upstream)

Reply:
It works 👍